File: app/src/main/java/com/my/notes/data/DataSource.java
package com.my.notes.data;

public interface DataSource {
        DataSource init(CardDataResponse cardDataResponse);
        CardData getData(int position);
        int size();
        void changeData(int position, CardData cardData);
        void addData(CardData cardData);
        CardData deleteData(int position);

    }

File: app/src/main/java/com/my/notes/data/DataSource.kt
package com.my.notes.data

interface DataSource {
    fun init(cardDataResponse: CardDataResponse): DataSource
    fun getData(position: Int): CardData
    fun size(): Int
    fun changeData(position: Int, cardData: CardData)
    fun addData(cardData: CardData)
    fun deleteData(position: Int): CardData
}